src/dune/dune.py

```diff
@@ -242,6 +230,27 @@ def list_nodes(self, simple=False, sep='|'):
         for state in states:
             print( state.string(sep=sep, simple=simple, name_width=name_width) )
 
+    def stop_conflicting_nodes(self, nod):
+        my_addrs=self._context.get_config_args(nod)
+        was_running=[]
+        was_active=None
+
+        # For each state, make decisions based on it's
+        for state in self.state_list():
+            # Don't operate on our node.
+            if state.name == nod.name():
+                continue
+            if state.is_active:
+                was_active = state.name
+            if state.is_running:
+                # We only need to stop a running node if there are address collisions.
+                if state.http in my_addrs or state.p2p in my_addrs or state.ship in my_addrs:
+                    was_running.append(state.name)
+                    self.stop_node(node(state.name))
```

Comment:
This was a mistake we made with the integration tests. It looks like `stop_node` does a `kill -9` which will cause the node to have a dirty db. If the user had a long running node that causes them to do a full replay or restore from snapshot. I think this should just fail to start with a good error message instead of killing the already running node.

Reply:
Thanks Kevin.

@larryk85 do you agree? If yes we should cancel this PR and issue. Then I would add new issue to change the logic so we don't stop conflicting nodes (currently it is done when starting up a node and exporting configs).

Reply:
[src/dune/dune.py#L168](https://github.com/AntelopeIO/DUNE/blob/mikelik/stop_conflicting_nodes/src/dune/dune.py#L168):
```python
         self._docker.execute_cmd(['kill', pid])
```

This looks like a simple `kill <pid>` which - I believe - defaults to SIGTERM. Maybe this is the wrong signal and should be SIGINT?

```python
     self._docker.execute_cmd(['kill', '-s SIGINT', pid])
```

Reply:
If it defaults to SIGTERM then that should be fine for not generating a dirty-db as long as you pause long enough for that to shutdown gracefully before terminating the docker container.  Regardless, if I have an existing running node seems like I would want that node to keep running and fail starting a new one, not shutdown the old one.